lib/hmmlearn/tests/test_variational_gaussian.py

```diff
@@ -265,33 +285,313 @@ def test_incorrect_init_covariance(self, implementation):
         assert np.all(model.dof_prior_ == 2.)
         assert np.all(model.scale_prior_ == 2.)
 
+        # Manually set everything
+        model = vhmm.VariationalGaussianHMM(
+            4, n_iter=500, tol=1e-9, random_state=random_state,
+            covariance_type=self.covariance_type,
+            implementation=implementation,
+            init_params="",
+        )
+        model.means_prior_ = [[0.], [0.], [0.], [0.]]
+        model.means_posterior_ = [[2], [1], [3], [4]]
+        model.beta_prior_ = [2., 2., 2., 2.]
+        model.beta_posterior_ = [1, 1, 1, 1]
+        model.dof_prior_ = [2., 2., 2., 2.]
+        model.dof_posterior_ = [1, 1, 1, 1]
+        modelscale_prior_ = [[[2.]], [[2.]], [[2.]], [[2.]]]
+        model.scale_posterior_ = [[[2.]], [[2.]], [[2.]], [[2.]]]
+        assert_log_likelihood_increasing(model, sequences, lengths, 10)
 
-# As other covariance types are implemented, refactor this.
-class NotImplementedYet:
 
-    covariance_type = None
+class TestTied(_TestGaussian):
+    test_fit_mcgrory_titterington1d_mean = 1.4774254
+    covariance_type = "tied"
 
     @pytest.mark.parametrize("implementation", ["scaling", "log"])
-    def test_fit_mcgrory_titterington(self, implementation):
+    def test_initialization(self, implementation):
         random_state = check_random_state(234234)
-        sequences, lengths = get_sequences(500, 1,
-                                           model=get_mcgrory_titterington())
+        sequences, lengths = get_sequences(
+            50, 10, model=get_mcgrory_titterington())
+
+        # dof's have wrong shape
+        with pytest.raises(ValueError):
+            model = vhmm.VariationalGaussianHMM(
```

Comment:
Instead of repeatedly setting up the same hmm except for the wrong attribute, perhaps factor that out in a new_hmm() method (as exists in other tests) which already sets all attributes to valid values, and just overwrite whatever needs to be in the pytest.raises() block?

Reply:
Great idea, thanks.